src/profile/profile.service.ts

```diff
@@ -63,6 +64,38 @@ export class ProfileService {
     }
   }
 
+  /**
+   * It updates a profile by id
+   * @param {string} profileId - string - The id of the profile to be deleted
+   * @param {UpdateProfileDto} updateProfileDto - UpdateProfileDto - the data to be updated
+   * @returns The delete method returns a DeleteResult object.
+   */
+
+  async updateProfile(
+    profileId: string,
+    updateProfileDto: UpdateProfileDto,
+  ): Promise<Profile> {
+    try{
+      const profile = await this.profileRepo.findOne({
+        where: { id: profileId },
+      });
+      if (profile) {
+        await this.profileRepo.update(profileId, updateProfileDto);
```

Comment:
I don't think you are supposed to update the profile Id

Reply:
@TijanAyo update function takes two arguments, criteria and dto, so that line is fine

Reply:
Ohh, that makes sense... I assumed it would also update the profile Id as well

Reply:
changes made